### DISCUSSION TITLE
Feature Request: Custom Sub-components with Checkbox Functionality in TanStack v8

### DISCUSSION THREAD

**Discussion by TsengTofu** [TanStack/table#5728] [Q&A]
Hi, I've tried some methods to achieve the goal as title.
Version: 8
## Table ColumnSettingList as below:
```javascript
// keep it simple
[
  {
    id: 'first_name',
    header: 'FirstName'
  },
  {
    id: 'age',
    header: 'Age',
  },
]
```

## Table Data structure as below:
If I want to use the api of "subRows", I need to have the same data structure as the parent row.
Then I could use the selection related api, every row, no matter is sub row or parent row.
What if I need in subRow is totally different header and column, but still need checkbox and relative with parent row?
```javascript
[
  {
    first_name: 'Andy',
    age: 20,
    subRows: [
      {
        first_name: 'Jerry',
        age: 40
      },
      {
        first_name: 'Jokie',
        age: 30
      },
    ],
  },
  {
    //...same structure
  },
]
```

## Goal:
Table can be expanded, selected all rows (child & parent), and custom subComponent.
Or it can be achieved by version 8?
Please help, thank you.

**Comment by KevinVandy** [comment]
You might be looking for making an entire sub table instead of sub rows

**Reply by TsengTofu** [reply]
Hello, thank you for your reply.
I'm curious about the definition of sub-rows and when to use them.
Clarify the difference between sub-rows with selection and sub-component without selection.
Do you have any methods to achieve this goal?

I've come up with an approach, but I'm not sure if it's feasible.
Instead of using the row selection API, I'm considering creating a custom hook for all checkboxes in the table.
Can you help me refine this idea or suggest alternatives?
